qiita_db/test/test_metadata_template.py

```diff
@@ -652,6 +775,133 @@ def test_create(self):
                ['2.Sample3', "Value for sample 3"]]
         self.assertEqual(obs, exp)
 
+    def test_create_int_prefix(self):
+        """Creates a new SampleTemplate"""
+        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
+        # The returned object has the correct id
+        self.assertEqual(st.id, 2)
+
+        # The relevant rows to required_sample_info have been added.
+        obs = self.conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
+        # sample_id study_id physical_location has_physical_specimen
+        # has_extracted_data sample_type required_sample_info_status_id
+        # collection_timestamp host_subject_id description
+        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
```

Comment:
I understand your concerns but not sure a global id is the actual way to go. For instance, if a user downloads 2 studies with the same sample names (blank.1, blank.2, ...) and tries to perform a meta-analysis, it will not work as expected as it will merge those samples. Having the idea of unique ids, will require the users to have an easy way to relate this unique id with their actual sample ids, which we do not have. The study_id.sample_id is not perfect but we haven't found a easier solution. Perhaps the way to go is to stop using text files and start using binary with extra metadata, like the hdf5 solutions we have talked about. Until those solutions are in place, I think this is our best bet. Perhaps, adding an issue so we do not forget and once we have solved the [next release](https://github.com/biocore/qiita/milestones/Alpha%200.1) issues we can address this and other ones in the same area.